src/reconcile/rec_write.c

```diff
@@ -1342,14 +1342,35 @@ __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
 		 * examining its updates. As prepared transaction id's are
 		 * globally visible, need to check the update state as well.
 		 */
-		if (F_ISSET(r, WT_REC_EVICT) &&
-		    (upd->prepare_state == WT_PREPARE_LOCKED ||
-		    upd->prepare_state == WT_PREPARE_INPROGRESS ||
-		    (F_ISSET(r, WT_REC_VISIBLE_ALL) ?
-		    WT_TXNID_LE(r->last_running, txnid) :
-		    !__txn_visible_id(session, txnid)))) {
-			uncommitted = r->update_uncommitted = true;
-			continue;
+		if (F_ISSET(r, WT_REC_EVICT)) {
+		       if (upd->prepare_state == WT_PREPARE_LOCKED ||
+			   upd->prepare_state == WT_PREPARE_INPROGRESS)
+			       prepared = true;
+
+		       if (F_ISSET(r, WT_REC_VISIBLE_ALL) ?
+			   WT_TXNID_LE(r->last_running, txnid) :
+			   !__txn_visible_id(session, txnid))
+			       uncommitted = r->update_uncommitted = true;
+
+		       /*
+			* TODO:
+			* The following portion of code under #ifdef is there
+			* to temporarily disable lookaside eviction of the
+			* prepared updates. Once we have all the pieces put
+			* together to enable the feature, remove this temporary
+			* code.
+			*/
+#define	DISABLE_LOOKASIDE_FOR_PREPARED
+#ifdef DISABLE_LOOKASIDE_FOR_PREPARED
+		       if (prepared) {
+			       prepared = false;
+			       uncommitted = r->update_uncommitted = true;
+		       }
+#endif
+#undef DISABLE_LOOKASIDE_FOR_PREPARED
+
+		       if (prepared || uncommitted)
+			       continue;
```

Comment:
Is it not required to reset the prepared to "false" before continue ?

Reply:
No, if we find an uncommitted or prepared update, we continue to not process the rest of the for loop because other than exceptions we do not see a stable update after an uncommitted and hence we wont find a suitable candidate for eviction.